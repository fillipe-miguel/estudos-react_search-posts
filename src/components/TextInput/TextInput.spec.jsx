import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { TextInput } from ".";

describe("<TextInput/>", () => {
    const fn = jest.fn();

    // NAO FUNFOU PQ O USER EVENT NÃO TA GUNCIONANDO LEGAL AQUI

    // it("should call handleChange function on each key pressed", async () => {
    //     render(<TextInput onChange={fn} value={"teste"} />);

    //     // Salvando o elemento dentro de uma variavel
    //     const element = screen.getByPlaceholderText(/Escreva sua pesquisa:/i);
    //     const value = "o valor";

    //     const user = userEvent.setup();

    //     // userEvent.type(element, value);

    //     await user.type(element, "es");

    //     // Vendo se o valor digitado ali é o valor do value
    //     expect(element.value).toBe("Teste");
    // });

    it("should have a search value of value", () => {
        render(<TextInput onChange={fn} value={"teste"} />);

        const element = screen.getByPlaceholderText(/Escreva sua pesquisa:/i);

        // testando se ele está renderizando
        // expect(element).toBeInTheDocument(); ==> faço esse teste por logica ali

        // testando se o valor passado ta realmente passando
        expect(element.value).toBe("teste");
    });

    it("should match with snappshot", () => {
        const { container } = render(
            <TextInput onChange={fn} value={"teste"} />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
