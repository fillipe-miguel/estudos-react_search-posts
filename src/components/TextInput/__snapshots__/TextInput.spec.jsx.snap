// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> should match with snappshot 1`] = `
<input
  class="text-input"
  placeholder="Escreva sua pesquisa:"
  type="search"
  value="teste"
/>
`;
